Form/View/ExportViewType.php

```diff
@@ -117,9 +138,32 @@ public function getBlockPrefix()
      */
     public function generateResponse(View $view, Request $request)
     {
-        $response = new Response();
         $parameters = $this->getParameters($view, $this->formFactory, $request);
-        
+
+        if (isset($view->getOptions()['export_type']) or $view->getOptions()['export_type'] === 'dompdf') {
+            // instantiate and use the dompdf class
+            $dompdf = new Dompdf();
+            $dompdf->loadHtml($parameters['render']);
+
+            // (Optional) Setup the paper size and orientation
+            $dompdf->setPaper(
+                (isset($view->getOptions()['pdf_size']) and $view->getOptions()['pdf_size']) ? $view->getOptions()['pdf_size'] : 'A4',
+                (isset($view->getOptions()['pdf_orientation']) and $view->getOptions()['pdf_orientation']) ? $view->getOptions()['pdf_orientation'] : 'portrait'
+            );
+
+            // Render the HTML as PDF
+            $dompdf->render();
+
+            // Output the generated PDF to Browser
+            $dompdf->stream($parameters['filename'] ?? "document.pdf", [
+                'compress' => 1,
+                'Attachment' => ( isset($view->getOptions()['disposition'])  && $view->getOptions()['disposition']  === 'attachment')?1:0,
+            ]);
+            exit;
```

Comment:
This function is doing multiple things, and to terminate in case of PDF generation, this exit is present?
I think the function should be split. Creation of a PDF should happen in it's own function, an exit statement should not be needed.

Reply:
Agree, the pdf generation should be relocated in a service